### PR TITLE
New intro docs

### PR DESCRIPTION
**Pull request by Datseris**
This is an attempt to make the documentation of Agents.jl more welcoming to new users. 

- Remove comparison from homepage
- Add some words about the design of agents in homepage
- Declutter tutorial by removing non-crucial information. (obviously the spaces are crucial information however listing there the docs of all spaces takes a lot of space, and is not necessary at the moment)

### CODE REVIEW

File: docs/src/api.md [hunk 1]
```diff
@@ -37,9 +52,21 @@ genocide!
 sample!
 ```
 
+### `@agents` macro
```
Comment (Libbum) [inline docs/src/api.md:55]: Should this section be moved to above "Adding agents"? Since this is ultimately "creating agents".

Comment (Datseris) [reply docs/src/api.md:55]: good point

File: docs/src/performance.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Performance Tips
```
Comment (Libbum) [inline docs/src/performance.md:1]: This isn't included anywhere. 

I guess we could move the 'a note on iteration' from the API page here though?

**Comment by Libbum** [comment]
These are good additions / changes. Whilst we're looking into this, one thing that always bothered me was that methods like `run!` were only on the tutorial page and not listed in the API.

**Comment by Datseris** [comment]
> Whilst we're looking into this, one thing that always bothered me was that methods like run! were only on the tutorial page and not listed in the API.

Unfortunately documenter doesn't allow expanding a documentation string in two places :(

**Comment by Libbum** [comment]
Oh really! That's... odd. 

Sure, then having a tutorial is more important.

File: docs/src/performance.md (hunk 2 above)
Comment (Libbum) [reply docs/src/performance.md:1]: This file is still here, but not linked anywhere...

**Pull request merged**